# HG changeset patch
# User Martin Stransky <stransky@redhat.com>
# Date 1526647470 -7200
#      Fri May 18 14:44:30 2018 +0200
# Branch vsync-non-x11
# Node ID 6d78f7e9a89df9a34b674acb5505274010b88cdd
# Parent  aabfe960ab59fea2e85896b1f8050786e16ab23b
Bug 1462622 - [Gtk/Linux] Don't use GLXVsyncSource() on non-X11 displays, r?lsalzman

MozReview-Commit-ID: BBtnNLWqSiq

diff -r aabfe960ab59 -r 6d78f7e9a89d gfx/thebes/gfxPlatformGtk.cpp
--- a/gfx/thebes/gfxPlatformGtk.cpp	Fri May 11 00:59:14 2018 +0300
+++ b/gfx/thebes/gfxPlatformGtk.cpp	Fri May 18 14:44:30 2018 +0200
@@ -741,8 +741,10 @@
   // Only use GLX vsync when the OpenGL compositor is being used.
   // The extra cost of initializing a GLX context while blocking the main
   // thread is not worth it when using basic composition.
+  // Also don't use it on non-X11 displays.
   if (gfxConfig::IsEnabled(Feature::HW_COMPOSITING)) {
-    if (gl::sGLXLibrary.SupportsVideoSync()) {
+    if (GDK_IS_X11_DISPLAY(gdk_display_get_default()) &&
+        gl::sGLXLibrary.SupportsVideoSync()) {
       RefPtr<VsyncSource> vsyncSource = new GLXVsyncSource();
       VsyncSource::Display& display = vsyncSource->GetGlobalDisplay();
       if (!static_cast<GLXVsyncSource::GLXDisplay&>(display).Setup()) {
